test/unit/mock/fake_runtime_client.py

```diff
@@ -451,6 +455,14 @@ def job_delete(self, job_id):
         self._get_job(job_id)
         del self._jobs[job_id]
 
+    def wait_for_final_state(self, job_id):
+        """Wait for the final state of a program job."""
+        final_states = ["COMPLETED", "FAILED", "CANCELLED", "CANCELLED - RAN TOO LONG"]
+        status = self._get_job(job_id).status()
+        while status not in final_states:
+            time.sleep(1)
```

Comment:
Is there any reason for putting in "1 second" here? We should try to keep the duration of our unit tests low and 1s is a long time...

Reply:
No particular reason, the original `wait_for_final_state` had a sleep but I can try removing it